### DISCUSSION TITLE
Is it possible to migrate from HiveDB to Isar and how?

### DISCUSSION THREAD

**Discussion by corusm** [isar/isar#84] [Help] [answered]
Hi there, can I start my App Development with HiveDB and swap later on to Isar?

**Comment by simc** [comment, marked as answer]
Yes you definitely can but there is no automatic migration. Hive supports storing deeply nested types and Isar requires you to use a more traditional schema.

**Reply by corusm** [reply]
How would the migration look like?
Must I write a dart script to migrate?

**Reply by simc** [reply]
Yes you need to do it yourself. I would suggest using the json import API of Isar.

**Reply by Sockitos** [reply]
@leisim What do you mean by 'json import API of Isar'? I can't seem to find it anywhere in isar's documentation or repository.

**Reply by simc** [reply]
Oh, maybe I forgot to write docs. There is a `collection.importJson()` method

**Reply by Sockitos** [reply]
Thanks! I probably missed it.
